i6_models/parts/frontend/vgg_act.py

```diff
@@ -0,0 +1,200 @@
+        pool2_padding: padding for second pooling layer
+        activation: activation function at the end
+        linear_input_dim: input size of the final linear layer
+        out_features: output size of the final linear layer
+    """
+
+    in_features: int
+    conv1_channels: int
+    conv2_channels: int
+    conv3_channels: int
+    conv4_channels: int
+    conv_kernel_size: IntTupleIntType
+    conv_padding: Optional[IntTupleIntType]
+    pool1_kernel_size: IntTupleIntType
+    pool1_stride: Optional[IntTupleIntType]
+    pool1_padding: Optional[IntTupleIntType]
+    pool2_kernel_size: IntTupleIntType
+    pool2_stride: Optional[IntTupleIntType]
+    pool2_padding: Optional[IntTupleIntType]
+    activation: Union[nn.Module, Callable[[torch.Tensor], torch.Tensor]]
+    linear_input_dim: Optional[int]
+    out_features: Optional[int]
+
+    def check_valid(self):
+        if isinstance(self.conv_kernel_size, int):
+            assert self.conv_kernel_size % 2 == 1, "ConformerVGGFrontendV1 only supports odd kernel sizes"
+        if isinstance(self.pool1_kernel_size, int):
+            assert self.pool1_kernel_size % 2 == 1, "ConformerVGGFrontendV1 only supports odd kernel sizes"
+        if isinstance(self.pool2_kernel_size, int):
+            assert self.pool2_kernel_size % 2 == 1, "ConformerVGGFrontendV1 only supports odd kernel sizes"
+
+    def __post__init__(self):
+        super().__post_init__()
+        self.check_valid()
+
+
+class VGG4LayerActFrontendV1(nn.Module):
+    """
+    Convolutional Front-End
+
+    The frond-end utilizes convolutional and pooling layers, as well as activation functions
+    to transform a feature vector, typically Log-Mel or Gammatone for audio, into an intermediate
+    representation.
+
+    Structure of the front-end:
+      - Conv
+      - Conv
+      - Activation
+      - Pool
+      - Conv
+      - Conv
+      - Activation
+      - Pool
+
+    Uses explicit padding for ONNX exportability, see:
+    https://github.com/pytorch/pytorch/issues/68880
+    """
+
+    def __init__(self, model_cfg: VGG4LayerActFrontendV1Config):
+        """
+        :param model_cfg: model configuration for this module
+        """
+        super().__init__()
+
+        model_cfg.check_valid()
+
+        conv_padding = (
+            model_cfg.conv_padding if model_cfg.conv_padding is not None else _get_padding(model_cfg.conv_kernel_size)
+        )
+        pool1_padding = model_cfg.pool1_padding if model_cfg.pool1_padding is not None else 0
+        pool2_padding = model_cfg.pool2_padding if model_cfg.pool2_padding is not None else 0
+
+        self.include_linear_layer = True if model_cfg.out_features is not None else False
+        assert not (
+            (model_cfg.linear_input_dim is not None) != (model_cfg.out_features is not None)
+        ), "please set input and output dim of the linear layer"
+
+        self.conv1 = nn.Conv2d(
+            in_channels=model_cfg.in_features,
```

Comment:
This is wrong, right?
```suggestion
            in_channels=1,
```